library/std/src/process.rs

```diff
@@ -1431,6 +1431,92 @@ impl From<fs::File> for Stdio {
     }
 }
 
+#[stable(feature = "stdio_from_stdio", since = "1.58.0")]
+impl From<io::Stdin> for Stdio {
```

Comment:
Questions about this impl &mdash; could you clarify whether this has any use that is not identical to `.stdin(Stdio::inherit())`?

I tried the following, but it didn't work as I'd naively expect:

```rust
use std::io::{self, Read};
use std::process::Command;

fn main() {
    let mut child = Command::new("whoami")
        .stdout(io::stdin())
        .spawn()
        .unwrap();

    let mut stdin = String::new();
    io::stdin().read_to_string(&mut stdin).unwrap();

    let status = child.wait().unwrap();
    assert!(status.success());

    println!("{:?}", stdin);
}
```

(I am on Linux.)

Instead of the child process's stdout feeding into the current process's stdin, the child's stdout goes to stdout and the current process stdin is unaffected.

The example code you gave in the documentation for this impl uses `.stdin(io::stdin())`. We would prefer that this always be written as `.stdin(Stdio::inherit())` though, correct?